ios/RNSFullWindowOverlay.mm

```diff
@@ -72,27 +101,55 @@ - (void)didMoveToWindow
     }
   } else {
     if (_touchHandler == nil) {
+#ifdef RN_FABRIC_ENABLED
+      _touchHandler = [RCTSurfaceTouchHandler new];
+#else
       _touchHandler = [[RCTTouchHandler alloc] initWithBridge:_bridge];
+#endif
     }
     [_touchHandler attachToView:_container];
   }
 }
 
+#ifdef RN_FABRIC_ENABLED
+#pragma mark - Fabric Specific
+
++ (facebook::react::ComponentDescriptorProvider)componentDescriptorProvider
+{
+  return facebook::react::concreteComponentDescriptorProvider<
+      facebook::react::RNSFullWindowOverlayComponentDescriptor>();
+}
+
+#else
+#pragma mark - Paper specific
+
 - (void)invalidate
```

Comment:
Probably the same should be done for Fabric in `prepareForRecycle`

Reply:
Resolved in call 
Also [1c5c4d9](https://github.com/software-mansion/react-native-screens/pull/1495/commits/1c5c4d9c859c9cf50223385c21ce09f0e88de174), [b559650](https://github.com/software-mansion/react-native-screens/pull/1495/commits/b559650014327da9d17e6d965da0e6b9b64547e0)